controllers/patient-controller.ts

```diff
@@ -0,0 +1,138 @@
+import { RequestHandler } from "express";
+import { Document } from "mongoose";
+
+import PatientModel, { IPatient } from "../models/PatientModel";
+import validationErrorHandler from "../utils/validation-error-handler";
+
+const getAllPatients: RequestHandler = async (req, res, next) => {
+  let patients;
+
+  try {
+    patients = await PatientModel.find({});
+  } catch (error) {
+    return res.status(500).json({ message: error.message });
+  }
+
+  return res.json({
+    patients: patients.map(patient => {
+      return patient.toObject({ getters: true });
+    })
+  });
+};
+
+const createPatient: RequestHandler = async (req, res, next) => {
+  const validationError = validationErrorHandler(req, res);
+
+  if (validationError) {
+    return validationError;
+  }
+
+  const {
+    fullName,
+    dob,
+    gender,
+    phoneNumber,
+    address,
+    zipCode,
+    maritalStatus,
+    job = ""
+  } = req.body;
+
+  let createdPatient = new PatientModel({
+    fullName,
+    dob,
+    gender,
+    phoneNumber,
+    address,
+    zipCode,
+    maritalStatus,
+    job,
+    history: {
+      chronic_diseases: "",
+      previous_admission: "",
+      past_surgery: "",
+      fractures: "",
+      family_history: "",
+      drug_allergy: "",
+      chronic_drug_usage: "",
+      smoking_status: "",
+      alcohol: "",
+      notes: ""
```

Comment:
I would think you can initialize and empty history object here, especially since those are not required fields